ts/src/pro/bitpanda.ts

```diff
@@ -0,0 +1,1335 @@
+            }
+            this.orders.append ({
+                'id': orderId,
+                'symbol': symbol,
+                'status': status,
+                'timestamp': this.parse8601 (datetime),
+                'datetime': datetime,
+            });
+        } else {
+            this.orders.append (this.parseOrder (update));
+        }
+        client.resolve (this.orders, 'orders');
+        // update balance
+        const balanceKeys = [ 'locked', 'unlocked', 'spent', 'spent_on_fees', 'credited', 'deducted' ];
+        for (let i = 0; i < balanceKeys.length; i++) {
+            const newBalance = this.safeValue (update, balanceKeys[i]);
+            if (newBalance !== undefined) {
+                this.updateBalance (newBalance);
+            }
+        }
+        client.resolve (this.balance, 'balance');
+        // update trades
+        if (updateType === 'TRADE_SETTLED') {
+            this.myTrades.append (this.parseTrade (update));
+            client.resolve (this.myTrades, 'myTrades');
+        }
+    }
+
+    parseWSOrderStatus (status) {
+        const statuses = {
+            'ORDER_REJECTED': 'rejected',
+            'ORDER_CLOSED': 'closed',
+            'STOP_ORDER_TRIGGERED': 'triggered',
+        };
+        return this.safeString (statuses, status, status);
+    }
+
+    updateBalance (balance) {
+        //
+        //     {
+        //         currency_code: 'EUR',
+        //         amount: '0.0',
+        //         new_available: '0.0',
+        //         new_locked: '0.15949533'
+        //     }
+        //
+        const currencyId = this.safeString (balance, 'currency_code');
+        const code = this.safeCurrencyCode (currencyId);
+        const account = this.account ();
+        account['free'] = this.safeString (balance, 'new_available');
+        account['used'] = this.safeString (balance, 'new_locked');
+        this.balance[code] = account;
+        this.balance = this.safeBalance (this.balance);
+    }
+
+    async watchOHLCV (symbol: string, timeframe = '1m', since: Int = undefined, limit: Int = undefined, params = {}) {
+        /**
+         * @method
+         * @name bitpanda#watchOHLCV
+         * @see https://developers.bitpanda.com/exchange/#candlesticks-channel
+         * @description watches historical candlestick data containing the open, high, low, and close price, and the volume of a market
+         * @param {string} symbol unified symbol of the market to fetch OHLCV data for
+         * @param {string} timeframe the length of time each candle represents
+         * @param {int|undefined} since timestamp in ms of the earliest candle to fetch
+         * @param {int|undefined} limit the maximum amount of candles to fetch
+         * @param {object} params extra parameters specific to the bitpanda api endpoint
+         * @returns {[[int]]} A list of candles ordered as timestamp, open, high, low, close, volume
+         */
+        await this.loadMarkets ();
```

Comment:
@pcriadoperez for some reason bitpand is not broadcasting candles
```
2023-05-15T11:46:14.863Z connecting to wss://streams.exchange.bitpanda.com with timeout 10000 ms
2023-05-15T11:46:15.079Z connected
2023-05-15T11:46:15.079Z ping loop
2023-05-15T11:46:15.080Z sending {'type': 'SUBSCRIBE', 'channels': [{'name': 'CANDLESTICKS', 'properties': [{'instrument_code': 'BTC_USDT', 'time_granularity': {'unit': 'MINUTES', 'period': 1}}]}]}
2023-05-15T11:46:15.132Z pong WSMessage(type=<WSMsgType.PONG: 10>, data=bytearray(b''), extra='')
2023-05-15T11:46:15.132Z message {"channels":[{"properties":[{"instrument_code":"BTC_USDT","time_granularity":{"unit":"MINUTES","period":1}}],"name":"CANDLESTICKS"}],"type":"SUBSCRIPTIONS","time":"2023-05-15T11:46:15.143408Z"}
2023-05-15T11:46:15.380Z sending {'type': 'UPDATE_SUBSCRIPTION', 'channels': [{'name': 'CANDLESTICKS', 'properties': [{'instrument_code': 'BTC_USDT', 'time_granularity': {'unit': 'MINUTES', 'period': 1}}, {'instrument_code': 'LTC_EUR', 'time_granularity': {'unit': 'MINUTES', 'period': 1}}]}]}
2023-05-15T11:46:15.436Z message {"channels":[{"properties":[{"instrument_code":"BTC_USDT","time_granularity":{"unit":"MINUTES","period":1}},{"instrument_code":"LTC_EUR","time_granularity":{"unit":"MINUTES","period":1}}],"name":"CANDLESTICKS"}],"type":"SUBSCRIPTIONS","time":"2023-05-15T11:46:15.445270Z"}
2023-05-15T11:46:15.436Z message {"channel_name":"CANDLESTICKS","type":"SUBSCRIPTION_UPDATED","time":"2023-05-15T11:46:15.445366Z"}
2023-05-15T11:46:25.080Z message {"subscription":"SYSTEM","channel_name":"SYSTEM","type":"HEARTBEAT","time":"2023-05-15T11:46:25.090120Z"}
2023-05-15T11:46:35.149Z message {"subscription":"SYSTEM","channel_name":"SYSTEM","type":"HEARTBEAT","time":"2023-05-15T11:46:35.089555Z"}
2023-05-15T11:46:45.079Z message {"subscription":"SYSTEM","channel_name":"SYSTEM","type":"HEARTBEAT","time":"2023-05-15T11:46:45.090015Z"}
2023-05-15T11:46:45.132Z pong WSMessage(type=<WSMsgType.PONG: 10>, data=bytearray(b''), extra='')
2023-05-15T11:46:55.120Z message {"subscription":"SYSTEM","channel_name":"SYSTEM","type":"HEARTBEAT","time":"2023-05-15T11:46:55.089517Z"}
2023-05-15T11:47:05.086Z message {"subscription":"SYSTEM","channel_name":"SYSTEM","type":"HEARTBEAT","time":"2023-05-15T11:47:05.089662Z"}
2023-05-15T11:47:15.092Z message {"subscription":"SYSTEM","channel_name":"SYSTEM","type":"HEARTBEAT","time":"2023-05-15T11:47:15.090249Z"}
2023-05-15T11:47:15.141Z pong WSMessage(type=<WSMsgType.PONG: 10>, data=bytearray(b''), extra='')
2023-05-15T11:47:25.081Z message {"subscription":"SYSTEM","channel_name":"SYSTEM","type":"HEARTBEAT","time":"2023-05-15T11:47:25.089497Z"}
2023-05-15T11:47:35.079Z message {"subscription":"SYSTEM","channel_name":"SYSTEM","type":"HEARTBEAT","time":"2023-05-15T11:47:35.089340Z"}
2023-05-15T11:47:45.079Z message {"subscription":"SYSTEM","channel_name":"SYSTEM","type":"HEARTBEAT","time":"2023-05-15T11:47:45.089621Z"}
2023-05-15T11:47:45.144Z pong WSMessage(type=<WSMsgType.PONG: 10>, data=bytearray(b''), extra='')
2023-05-15T11:47:55.123Z message {"subscription":"SYSTEM","channel_name":"SYSTEM","type":"HEARTBEAT","time":"2023-05-15T11:47:55.090032Z"}
2023-05-15T11:48:05.161Z message {"subscription":"SYSTEM","channel_name":"SYSTEM","type":"HEARTBEAT","time":"2023-05-15T11:48:05.089720Z"}
2023-05-15T11:48:15.099Z message {"subscription":"SYSTEM","channel_name":"SYSTEM","type":"HEARTBEAT","time":"2023-05-15T11:48:15.089322Z"}
2023-05-15T11:48:15.155Z pong WSMessage(type=<WSMsgType.PONG: 10>, data=bytearray(b''), extra='')
```

Reply:
it's very strange, it works for me about 2/10 times

Reply:
@pcriadoperez now I got a different error 
```
2023-05-16T09:01:49.923Z connecting to wss://streams.exchange.bitpanda.com with timeout 10000 ms
2023-05-16T09:01:50.139Z connected
2023-05-16T09:01:50.139Z ping loop
2023-05-16T09:01:50.139Z sending {'type': 'SUBSCRIBE', 'channels': [{'name': 'CANDLESTICKS', 'properties': [{'instrument_code': 'BTC_USDT', 'time_granularity': {'unit': 'MINUTES', 'period': 1}}]}]}
2023-05-16T09:01:50.193Z pong WSMessage(type=<WSMsgType.PONG: 10>, data=bytearray(b''), extra='')
2023-05-16T09:01:50.193Z message {"channels":[{"properties":[{"instrument_code":"BTC_USDT","time_granularity":{"unit":"MINUTES","period":1}}],"name":"CANDLESTICKS"}],"type":"SUBSCRIPTIONS","time":"2023-05-16T09:01:50.213721Z"}
2023-05-16T09:01:50.439Z sending {'type': 'UPDATE_SUBSCRIPTION', 'channels': [{'name': 'CANDLESTICKS', 'properties': [{'instrument_code': 'BTC_USDT', 'time_granularity': {'unit': 'MINUTES', 'period': 1}}, {'instrument_code': 'BTC_EUR', 'time_granularity': {'unit': 'MINUTES', 'period': 1}}]}]}
2023-05-16T09:01:50.492Z message {"channels":[{"properties":[{"instrument_code":"BTC_USDT","time_granularity":{"unit":"MINUTES","period":1}},{"instrument_code":"BTC_EUR","time_granularity":{"unit":"MINUTES","period":1}}],"name":"CANDLESTICKS"}],"type":"SUBSCRIPTIONS","time":"2023-05-16T09:01:50.510524Z"}
2023-05-16T09:01:50.492Z message {"channel_name":"CANDLESTICKS","type":"SUBSCRIPTION_UPDATED","time":"2023-05-16T09:01:50.510592Z"}
2023-05-16T09:01:50.510Z message {"instrument_code":"BTC_EUR","granularity":{"unit":"MINUTES","period":1},"high":"24974.5","low":"24974.5","open":"24974.5","close":"24974.5","volume":"12.48725","last_sequence":2864885,"channel_name":"CANDLESTICKS","type":"CANDLESTICK_SNAPSHOT","time":"2023-05-16T09:01:59.999999Z"}
Exchange.filter_by_since_limit() takes from 2 to 5 positional arguments but 6 were given
```

Reply:
nvm I had an outdated version apparently